tests/validations/test_saved_query.py

```diff
@@ -38,6 +38,21 @@ def check_only_one_error_with_message(  # noqa: D
     } and found_match
 
 
+def check_only_one_warning_with_message(  # noqa: D
```

Comment:
Based on the `check_onlye_one_error_with_message` logic above.  I thought it would make more sense to factor this out and make it reusable than to inline it.